rebasehelper/checkers/license_tool.py

```diff
@@ -0,0 +1,99 @@
+# -*- coding: utf-8 -*-
+#
+# This tool helps you to rebase package to the latest version
+# Copyright (C) 2013-2014 Red Hat, Inc.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# he Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Authors: Petr Hracek <phracek@redhat.com>
+#          Tomas Hozza <thozza@redhat.com>
+
+import os
+
+from six import StringIO
+
+from rebasehelper.utils import ProcessHelper
+from rebasehelper.checker import BaseChecker
+
+
+class LicenseTool(BaseChecker):
+    """license compare tool"""
+
+    CMD = "licensecheck"
+    DEFAULT = True
+    category = "SOURCE"
+
+    @classmethod
+    def match(cls, cmd=None):
+        if cmd == cls.CMD:
+            return True
+        else:
+            return False
+
+    @classmethod
+    def get_checker_name(cls):
+        return cls.CMD
+
+    @classmethod
+    def is_default(cls):
+        return cls.DEFAULT
+
+    @classmethod
+    def get_license_changes(cls):
+        """
+        Finds out if there is a possibility of license change in new sources
+
+        :return: true if license changes probably happened
+        """
+        old_sources_path = os.path.join(cls.workspace_dir, 'old_sources')
+        new_sources_path = os.path.join(cls.workspace_dir, 'new_sources')
+
+        diffs = []
+        for source_dir in [old_sources_path, new_sources_path]:
+            out = StringIO()
+            ProcessHelper.run_subprocess(["/usr/bin/licensecheck", source_dir, "--machine", "--recursive"],
+                                         output_file=out)
+            diff = set()
+            for l in out:
+                diff.add(l.split('\t')[1])
+            diffs.append(diff)
+
+        for new_licenses in diffs[1]:
+            if new_licenses not in diffs[0]:
+                cls.license_changes = True
```

Comment:
Ok, this only checks if some specific license newly appeared. I don't think that's enough. While it shouldn't matter when some file changes license to some different but already present license, I think it's worth mentioning to a maintaner.

Actually, this tool should ideally make list of all files which changed license, group it by specific license transitions and save it as a report.

Reply:
You are right, I will modify it.

Reply:
So far the output looks like this on fcitx-qt5 package:
```
licensecheck
======
License changes occured!
Check the following files:
/home/skisela/fedora/fcitx-qt5/rebase-helper-workspace/new_sources/fcitx-qt5-1.2.2/platforminputcontext/qfcitxplatforminputcontext.cpp license change:GPL (v2 or later) => UNKNOWN
/home/skisela/fedora/fcitx-qt5/rebase-helper-workspace/new_sources/fcitx-qt5-1.2.2/platforminputcontext/main.h license change:GPL (v2 or later) => UNKNOWN
/home/skisela/fedora/fcitx-qt5/rebase-helper-workspace/new_sources/fcitx-qt5-1.2.2/guiwrapper/main.cpp license change:GPL (v2 or later) => GPL (v3 or later)
/home/skisela/fedora/fcitx-qt5/rebase-helper-workspace/new_sources/fcitx-qt5-1.2.2/platforminputcontext/main.cpp license change:GPL (v3 or later) => UNKNOWN
/home/skisela/fedora/fcitx-qt5/rebase-helper-workspace/new_sources/fcitx-qt5-1.2.2/platforminputcontext/main.cpp license change:GPL (v2 or later) => UNKNOWN
/home/skisela/fedora/fcitx-qt5/rebase-helper-workspace/new_sources/fcitx-qt5-1.2.2/quickphrase-editor/main.h license change:GPL (v2 or later) => GPL (v3 or later)
/home/skisela/fedora/fcitx-qt5/rebase-helper-workspace/new_sources/fcitx-qt5-1.2.2/quickphrase-editor/main.cpp license change:GPL (v2 or later) => GPL (v3 or later)
/home/skisela/fedora/fcitx-qt5/rebase-helper-workspace/new_sources/fcitx-qt5-1.2.2/platforminputcontext/qfcitxplatforminputcontext.h license change:GPL (v2 or later) => UNKNOWN
/home/skisela/fedora/fcitx-qt5/rebase-helper-workspace/new_sources/fcitx-qt5-1.2.2/COPYING.BSD license change:New file => BSD (3 clause)
```

Reply:
Where `New file => ${some_license}` means a new file appeared and `${some_license} => UNKNOWN` means it does not appear in the new sources. I guess substituting it with "deleted" is more suitable.
Also a newline between file_path and license change is appropriate.

Reply:
I guess it is worth mentioning only the changed licenses, not the deleted+added(not new ones)

Reply:
No, I think additions and deletions should be included as well, but perhaps in a different way than changes. Maintainer needs to know whether there is a possibility that the `License` tag in a spec file requires adjustment.

And the output should be grouped by license transition, to make it more compact and readable.